mk/compile.mk

```diff
@@ -230,4 +230,57 @@ endef
 
 $(foreach f,$(asm-defines-files),$(eval $(call gen-asm-defines-file,$(f),$(out-dir)/$(sm)/include/generated/$(basename $(notdir $(f))).h)))
 
+# Device tree source file compilation
+DTC 	:= dtc
+DTC_FLAGS += -I dts -O dtb
+DTC_FLAGS += -Wno-unit_address_vs_reg
+
+define gen-dtb-file
+# dts file path/name in $1
+# dtb file path/name in $2
+
+dtb-predts-$2 	:= $$(dir $2)$$(notdir $2).pre.dts
+dtb-predep-$2 	:= $$(dir $2)$$(notdir $2).o.d
+dtb-dep-$2	:= $$(dir $2)$$(notdir $2).d
+dtb-cmd-file-$2	:= $$(dir $2).$$(notdir $2).cmd
+
+cleanfiles := $$(cleanfiles) $2 \
+		$$(dtb-predts-$2) $$(dtb-predep-$2) \
+		$$(dtb-dep-$2) $$(dtb-cmd-file-$2)
+
+dtb-cppflags-$2	:= -Icore/include/ -x assembler-with-cpp \
+		   -E -ffreestanding $$(CPPFLAGS) \
+		   -MD -MF $$(dtb-predep-$2) -MT $2
+
+dtb-dtcflags-$2	:= $$(DTC_FLAGS) -d $$(dtb-dep-$2)
+
+-include $$(dtb-dep-$2)
+-include $$(dtb-predep-$2)
+-include $$(dtb-cmd-file-$2)
+
+dtb-precmd-$2 = $$(CPP$(sm)) $$(dtb-cppflags-$2) -o $$(dtb-predts-$2) $$<
+dtb-cmd-$2 = $$(DTC) $$(dtb-dtcflags-$2) -o $$@ $$(dtb-predts-$2)
+
+$2: $1 FORCE
```

Comment:
I am unsure about this `FORCE`. Required?

Reply:
Yes, it's needed since you're keeping track of changes in the command line.

Reply:
Ok, as I understand, we enforces the rule resolution so that we go through the `$$(if $$(strip <cond>),<cmd>)` that takes care of deps resolution. Am I right?
If right, I see nothing to remove.